crates/router/src/core/payments/operations/payment_status.rs

```diff
@@ -399,14 +399,20 @@ async fn get_tracker_for_sync<
         .to_not_found_response(errors::ApiErrorResponse::BusinessProfileNotFound {
             id: profile_id.to_string(),
         })?;
+
     let payment_method_info =
         if let Some(ref payment_method_id) = payment_attempt.payment_method_id.clone() {
-            Some(
-                db.find_payment_method(payment_method_id)
-                    .await
-                    .to_not_found_response(errors::ApiErrorResponse::PaymentMethodNotFound)
-                    .attach_printable("error retrieving payment method from DB")?,
-            )
+            match db.find_payment_method(payment_method_id).await {
+                Ok(payment_method) => Some(payment_method),
+                Err(error) => {
+                    if error.current_context().is_db_not_found() {
+                        logger::info!("Error retrieving payment method from DB {:?}", error);
+                        None
```

Comment:
Is this error log required? If required then change it to something like
`Payment Method was not found in db`. It's not an error from db right?

Reply:
Yes, it is required log. Sure I will change the error message.